azure-ipam/ipam_test.go

```diff
@@ -227,15 +385,25 @@ func TestCmdDel(t *testing.T) {
 
 	tests := []scenario{
 		{
-			name:    "Happy CNI del",
-			args:    buildArgs("happyArgs", happyPodArgs, happyNetConfByteArr),
+			name:    "Happy CNI del single IP",
+			args:    buildArgs("happyArgsSingle", happyPodArgs, happyNetConfByteArr),
+			wantErr: false,
+		},
+		{
+			name:    "Happy CNI del dual IP",
+			args:    buildArgs("happyArgsDual", happyPodArgs, happyNetConfByteArr),
 			wantErr: false,
 		},
 		{
-			name:    "Fail request CNS release IP during CmdDel",
+			name:    "Fail request CNS release using ReleaseIPAddress during CmdDel",
 			args:    buildArgs("failRequestCNSReleaseIPArgs", happyPodArgs, happyNetConfByteArr),
 			wantErr: true,
 		},
+		{
+			name:    "Fail request CNS release during CmdDel",
+			args:    buildArgs("failRequestCNSReleaseIPsArgs", happyPodArgs, happyNetConfByteArr),
```

Comment:
this case should validate if azure-ipam calls ReleaseIPAddress api if ReleaseIPs fails

Reply:
Right now in the code the only time that we call `ReleaseIPAddress` is if we get a 404 error when we call `ReleaseIPs`. 

With the four tests that exist now the first happy path uses  `ReleaseIPs` and works fine since it is the happy path, second one first calls `ReleaseIPs`, fails with a 404 error, and then calls `ReleaseIPAddress` which passes, third fails does the same as second but fails at `ReleaseIPAddress`, and the fourth fails `ReleaseIPs` with an error that is not 404 so we don't call `ReleaseIPAddress`.